woocommerce/wc-admin
# show Pinterest for WooCommerce in `Marketing > Installed`

This PR adds a new row to `Marketing > Installed` table for (forthcoming) Pinterest for WooCommerce extension. This provides extra guidance and support to merchants to get Pinterest set up and working for their store.

<img width="1721" alt="Screen Shot 2021-07-27 at 3 14 30 PM" src="https://user-images.githubusercontent.com/4167300/127089627-34feecfe-f93d-48d6-854a-b90fa957411b.png">

### Accessibility

This is using same components / UI as the other rows, so should be equally accessible (I haven't tested this myself).

### Detailed test instructions:

_Prerequisite_: This requires the forthcoming native [__Pinterest for WooCommerce__ extension](https://github.com/saucal/pinterest-for-woocommerce) (private repo – may not be available). Ping me if you want a zip of Pinterest to test with!

If __Pinterest for WooCommerce__ is installed, marketing screen should show status info.

1. Install __Pinterest for WooCommerce__ extension. Don't activate it.
1. Go to `WooCommerce > Marketing`.
1. Confirm there is a row for Pinterest in `Installed marketing extensions` with `Activate` button.
1. Activate the extension by clicking the button, or via normal `WordPress > Plugins` screen.
1. Go to `WooCommerce > Marketing`.
1. Confirm the Pinterest row guides merchant to set up the extension (if not yet set up) or links to settings and docs.
2. When onboarding is complete, confirm the row has links to settings, docs, etc consistent with the other marketing extensions.

Marketing screen should be unaffected if __Pinterest for WooCommerce__ is not installed.

1. Ensure __Pinterest for WooCommerce__ is not installed.
1. Go to `WooCommerce > Marketing`.
1. Should look and work the same as in previous versions.

### ~~This PR is blocked - do not merge~~ Ready to review!
~~There's one blocker - the settings UI (and URL) is still in development in the Pinterest extension. When that's wrapped up I'll revisit this PR and fix the following things:~~

- [x] Use the correct URL for `Settings` link. **Note** the docs URL is not currently live/final, but it's highly likely to use the URL in the code so let's merge so we don't have to revisit later.
- [x] Detect whether setup (onboarding) is completed, and if not, guide merchant to set up Pinterest for WooCommerce with `Finish setup` button (consistent with other extensions).
